### PR TITLE
java-ipfs article

### PR DESCRIPTION
**Pull request by gjeanmart**

### CODE REVIEW

File: java/java-ipfs.md [hunk 1]
```diff
@@ -0,0 +1,489 @@
+# IPFS and Java
```
Comment (ChrisChinchilla) [inline java/java-ipfs.md]:
```suggestion
# Managing storage in a Java application with IPFS
```

Something like that?

File: java/java-ipfs.md [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+java version "1.8.0_201"
+```
+
+- A package and dependency manager, for example [Maven](https://maven.apache.org) or [Gradle](https://gradle.org)
+
+- An IDE (Integrated development environment), for this tutorial, we use Eclipse
+
+- A running IPFS node (> 0.4.x)
+*Follow [the following article](https://kauri.io/article/b01b9b7bebcd4ebf80edf021bdd0e232/v2/installing-ipfs) to learn how to install an IPFS node (go-ipfs)*
+```shell
```
Comment (ChrisChinchilla) [inline java/java-ipfs.md]: As there's a few different ways to start a node, and you also need to run `ipfs init` first anyway, I'd say, just remove this code snippet and leave the link instead for more information.

File: java/java-ipfs.md [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+
+## Dependencies
+
+To get started, it is required to import the `java-ipfs-http-client` dependency 
+
+### Maven
+
+Using Maven, we first need to configure the repository where the dependency is hosted and then import the dependency.
+
+*[pom.xml](https://github.com/gjeanmart/kauri-content/blob/master/java-ipfs/pom.xml)*
```
Comment (ChrisChinchilla) [inline java/java-ipfs.md]: Without this line, this doesn't work, so should you add it?

```
   <properties>
        <maven.compiler.target>1.8</maven.compiler.target>
        <maven.compiler.source>1.8</maven.compiler.source>
        <java-ipfs-http-client.version>v1.2.3</java-ipfs-http-client.version>
    </properties>
```

**Review by ChrisChinchilla — CHANGES_REQUESTED** [review]
@gjeanmart Started making some suggestions…

File: java/java-ipfs.md (hunk 3 above)
Comment (gjeanmart) [reply java/java-ipfs.md]: agree

File: java/java-ipfs.md [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+
+
+### Resolve
+
+Just like a DNS, reading an Object from an IPNS name is a two steps process:
+1. Resolve the hash against the name
+2. Read the content from the hash
+
+```java
+String resolveResponse = ipfs.name.resolve(key.id);
```
Comment (ChrisChinchilla) [inline java/java-ipfs.md]: @gjeanmart Should this code example be in a loop or something? i.e. what is `key` in this case?

File: java/java-ipfs.md [hunk 5]
```diff
@@ -0,0 +1,485 @@
...
+
+
+### Node peers
+
+To retrieve the list of Peers connected to your local node:
+
+
+```java
+List<Multihash> peers = ipfs.refs.local()
+peers.refs.local().forEach(multihash ->
```
Comment (ChrisChinchilla) [inline java/java-ipfs.md]: @gjeanmart Should this be

```suggestion
peers.forEach(multihash ->
```

File: java/java-ipfs.md (hunk 4 above)
Comment (gjeanmart) [reply java/java-ipfs.md]: `key` is the key generated at the step "Generate a key". we can indeed get the key by name via a loop
```java
try {
  KeyInfo key = localIPFS.key.list().stream()
    .filter(k -> k.name.equals(keyName))
    .findAny()
    .orElseThrow(() -> new RuntimeException("Key " + keyName + " not found"));

  String resolveResponse = ipfs.name.resolve(key.id);
  System.out.println("resolve(key="+key.id+"): " + resolveResponse);

  byte[] content = ipfs.cat(Multihash.fromBase58(resolveResponse.substring(6)));
  System.out.println("Content: " + new String(content));
} catch (IOException ex) {
 throw new RuntimeException("Error whilst communicating with the IPFS node", ex);
}

**Comment by gjeanmart** [comment]
Looks good to me. I'm merging the PR

**Pull request merged**